### PR TITLE
Why: Feature to deploy monit on target host and make the module work with newer versions of monit

### PR DESCRIPTION
**Pull request by prakritish**
##### SUMMARY

Feature to deploy monit on target host and fix functionalities broken due to changed output in newer versions of monit.

##### ISSUE TYPE

 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME

monit
##### ANSIBLE VERSION

```
[root@prakritish1 ansible]# ansible --version
ansible 2.2.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
[root@prakritish1 ansible]#
```

##### ADDITIONAL INFORMATION

These changes adds new functionalities as well as solve a few bugs:
###### New Functionalities

1. Allows users to deploy/install **monit** on the target host by specifying either the _version_ or _URL_ for the **monit** binary. This does not use packaged distribution, but downloads directly from **monit** distribution website. The module identifies the platform and architecture of the target host and downloads the relevant binary and deploys them at the specified path.

1. Adds capabilities to specify the _**path**_ of the **monit** binary and it's _**config**_ file if they are not available in default path.

1. Added _service_ as an alias for _name_.

###### Bug Fix
With the last few versions of **monit**, the output of the binary was changed and this may have broken the module and none of the existing functionalities were working. So, made the changes to fix the broken functionalities.

```

```

### CODE REVIEW

File: lib/ansible/modules/monitoring/monit.py [hunk 1]
```diff
@@ -23,26 +23,31 @@
...
 
-
 DOCUMENTATION = '''
 ---
 module: monit
 short_description: Manage the state of a program monitored via Monit
 description:
      - Manage the state of a program monitored via I(Monit)
-version_added: "1.2"
+version_added: "1.3"
```
Comment (bcoca) [inline lib/ansible/modules/monitoring/monit.py]: not sure why you are changing version here

File: lib/ansible/modules/monitoring/monit.py [hunk 2]
```diff
@@ -23,26 +23,31 @@
...
+    required: false
     default: null
+    version_added: "2.4"
   state:
     description:
       - The state of service
     required: true
     default: null
-    choices: [ "present", "started", "stopped", "restarted", "monitored", "unmonitored", "reloaded" ]
+    choices: [ "present", "started", "stopped", "restarted", "monitored", "unmonitored", "reloaded", "deployed" ]
```
Comment (bcoca) [inline lib/ansible/modules/monitoring/monit.py]: I don't think installation should be covered by this modules, there are plenty of package management modules that are better suited for this, at worst a role that downloads and installs it

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/monitoring/monit.py (hunk 1 above)
Comment (prakritish) [reply lib/ansible/modules/monitoring/monit.py]: Reverted back to "1.2" as suggested.

File: lib/ansible/modules/monitoring/monit.py (hunk 2 above)
Comment (prakritish) [reply lib/ansible/modules/monitoring/monit.py]: Removed "deployed" and the implementation as suggested.

**Comment by prakritish** [comment]
The requested changes are in place. Please review and let me know if any further changes are needed. 

**Playbook**
```
- hosts: monitHost
  become: yes
  become_method: sudo
  vars:
    version: 5.21.0
  tasks:
  - name: Start monit if not started
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      state: started
  - name: Check service {{ ohai_machinename }} present
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      service: "{{ ohai_machinename }}"
      state: "present"
  - name: Start service apache
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      name: "apache"
      state: "started"
  - name: Reload monit configuration
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      state: "reloaded"
  - name: Stop Service apache
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      service: "apache"
      state: "stopped"
  - name: Restart apache
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      service: "apache"
      state: "restarted"
  - name: Unmonitor Service apache
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      name: "apache"
      state: "unmonitored"
  - name: Monitor Service apache
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      service: "apache"
      state: "monitored"
  - name: Stop monit
    monit:
      path: "/opt/monit/bin/monit"
      config: "/etc/monitrc"
      state: "stopped"
```
**Output:**
```
[root@prakritish1 playbooks]# ansible-playbook monit.yml

PLAY [monitHost] ***************************************************************

TASK [setup] *******************************************************************
ok: [prakritish3]

TASK [Start monit if not started] **********************************************
changed: [prakritish3]

TASK [Check service prakritish3 present] ***********************
ok: [prakritish3]

TASK [Start service apache] ****************************************************
ok: [prakritish3]

TASK [Reload monit configuration] **********************************************
changed: [prakritish3]

TASK [Stop Service apache] *****************************************************
changed: [prakritish3]

TASK [Restart apache] **********************************************************
changed: [prakritish3]

TASK [Unmonitor Service apache] ************************************************
changed: [prakritish3]

TASK [Monitor Service apache] **************************************************
changed: [prakritish3]

TASK [Stop monit] **************************************************************
changed: [prakritish3]

PLAY RECAP *********************************************************************
prakritish3 : ok=10   changed=7    unreachable=0    failed=0

[root@prakritish1 playbooks]#
```

**Pull request closed without merging**